src/Catel.MVVM/Catel.MVVM.Shared/Collections/SuspensionContext.cs

```diff
@@ -128,33 +128,22 @@ public List<T> OldItems
         /// <summary>
         /// Tries to remove the item from old items
         /// </summary>
-        /// <param name="index">
-        /// The item index.
-        /// </param>
         /// <param name="item">
         /// The item.
         /// </param>
         /// <returns>
         /// <c>true</c> if removed, otherwise <c>false</c>.
         /// </returns>
-        public bool TryRemoveItemFromOldItems(int index, T item)
+        public bool TryRemoveItemFromOldItems(T item)
         {
             if (Mode == SuspensionMode.None || Mode == SuspensionMode.Mixed)
             {
                 var oldIdx = OldItems.LastIndexOf(item);
```

Comment:
List.LastIndexOf has O(n). As this method is called upon .Add which has O(1) this results in O(n) for adding one item.